### PR TITLE
[DistDGL][TestCoverage]Added testcase for testing distributed lookup service.

### PR DESCRIPTION
**Pull request by kylasa**
Added a new CI testcase for testing distributed lookup service. 

Distributed lookup service, today is used to retrieve partition-id for a given global-nid. 
By definition it is a distributed service and all the processes participate in this service. 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [X] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [X] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [X The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [X] All changes have test coverage
- [X] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by Rhett-Ying** [comment]
@kylasa  CI is failing yet, pls fix it first.

File: tests/tools/test_dist_lookup.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+
+@pytest.mark.parametrize(
+    "num_chunks, num_parts, world_size",
+    [[4, 4, 4], [8, 4, 2], [8, 4, 4], [9, 6, 3], [11, 11, 1], [11, 4, 1]],
+)
+def test_lookup_service(
+    num_chunks,
+    num_parts,
+    world_size,
+    num_chunks_nodes=None,
```
Comment (Rhett-Ying) [inline tests/tools/test_dist_lookup.py:202]: will we test these arguments in the near future? If not, let's remove them from here. Just specify `None` when calling `create_chunked_dataset()`.

**Comment by Rhett-Ying** [comment]
@kylasa  pls pay attention to below error log:
```
Traceback (most recent call last):

  File "/root/jenkins/workspace/dgl_PR-5365/tools/distpartitioning/data_proc_pipeline.py", line 9, in <module>

    from data_shuffle import multi_machine_run, single_machine_run

  File "/root/jenkins/workspace/dgl_PR-5365/tools/distpartitioning/data_shuffle.py", line 18, in <module>

    from dist_lookup import DistLookupService

  File "/root/jenkins/workspace/dgl_PR-5365/tools/distpartitioning/dist_lookup.py", line 12, in <module>

    from tools.distpartitioning.utils import map_partid_rank, memory_snapshot

ModuleNotFoundError: No module named 'tools'

```

**Comment by kylasa** [comment]
PATH=/home/ubuntu/software/mpi-install/bin:$PATH
export DGL_HOME=/home/ubuntu/software/dgl
export DGL_LIBRARY_PATH=$DGL_HOME/build:.:..
export PYTHONPATH=${PYTHONPATH}:.:..:$DGL_HOME/tools:$DGL_HOME/tools/distpartitioning:$DGL_HOME/tests:$DGL_HOME/python:/usr/local/lib


This is what I use in the .profile on my CI test machine. 
Please change the python path accordingly on the CI machine ...

File: tests/tools/test_dist_lookup.py (hunk 1 above)
Comment (kylasa) [reply tests/tools/test_dist_lookup.py:202]: This functionality will be needed when we remove the dependency between no. of files for edges and no. of files for edge features. After we support this request, there can be any no. of edge feature files.

**Comment by Rhett-Ying** [comment]
@kylasa you're responsible for fixing path issue as this PR introduces such dependency. pls refer to https://github.com/dmlc/dgl/blob/1313067e9a39389ed6246ea1d258bfa917856b84/tests/scripts/task_distributed_test.sh#L39 We don't need to make any direct change on CI machines.

**Review by Rhett-Ying — APPROVED** [review]
LGTM

**Pull request merged**